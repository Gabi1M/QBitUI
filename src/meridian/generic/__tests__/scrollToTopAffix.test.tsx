import React from 'react';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

const mockScrollTo = jest.fn();

jest.mock('@mantine/hooks', () => ({
    ...jest.requireActual('@mantine/hooks'),
    useWindowScroll: jest.fn().mockReturnValue([{ y: 10 }, mockScrollTo]),
}));

import ScrollToTopAffix from '../scrollToTopAffix';

describe('ScrollToTopAffix', () => {
    beforeEach(() => {
        mockScrollTo.mockReset();
    });

    it('should render as expected', () => {
        const result = render(<ScrollToTopAffix />);
        expect(result.asFragment()).toMatchSnapshot();
    });

    it('should call scroll to top on click', async () => {
        render(<ScrollToTopAffix />);
        await userEvent.click(screen.getByTestId('affixButton'));
        expect(mockScrollTo).toHaveBeenCalled();
    });
});
