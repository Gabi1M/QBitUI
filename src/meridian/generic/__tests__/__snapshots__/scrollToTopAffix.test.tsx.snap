// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTopAffix should render as expected 1`] = `<DocumentFragment />`;
